src/Spectre.Service/Loaders/DatasetLoader.cs

```diff
@@ -0,0 +1,136 @@
+﻿/*
+ * DatasetLoader.cs
+ * Class loading datasets from either local root or remote root.
+ *
+   Copyright 2017 Dariusz Kuchta
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using Ninject;
+using Spectre.Data.Datasets;
+using Spectre.Dependencies;
+using Spectre.Service.Configuration;
+
+namespace Spectre.Service.Loaders
+{
+    /// <summary>
+    /// Class for loading datasets from specified directories.
+    /// </summary>
+    public class DatasetLoader
+    {
+        #region Fields
+
+        /// <summary>
+        /// Root for local directory.
+        /// </summary>
+        private readonly string _localRoot;
+
+        /// <summary>
+        /// Root for remote directory.
+        /// </summary>
+        private readonly string _remoteRoot;
+
+        #endregion
+
+        #region Constructor
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DatasetLoader"/> class.
+        /// </summary>
+        /// <param name="dataRootConfig">Validated configuration of needed directories.</param>
+        public DatasetLoader(DataRootConfig dataRootConfig)
+        {
+            _localRoot = dataRootConfig.LocalPath;
+            _remoteRoot = dataRootConfig.RemotePath;
+
+            FileSystem = DependencyResolver.GetService(typeof(IFileSystem)) as IFileSystem;
+        }
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Handle to file system.
+        /// </summary>
+        public IFileSystem FileSystem { private get; set; }
```

Comment:
Shouldn't it be `readonly`? Why `set` is `public`? Especially when `get` is `private`? It is assumed to be an antipattern. I thought we are switching this only in the tests, before any call to kernel is done.